### ISSUE TITLE
field like "__xxx__" is not fixed correctly.

### ISSUE THREAD

**Issue by jjeffcaii** [palantir/tslint#2965]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__:  CLI

#### TypeScript code being linted

```ts
// code snippet
const obj = {};
obj["__id__"] = "foobar";
// then execute: tslint -c tslint.json 'src/**/*.ts' --fix

```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended",
  "rules": {
    "interface-name": false,
    "new-parens": true,
    "no-arg": true,
    "no-bitwise": true,
    "no-conditional-assignment": true,
    "no-shadowed-variable": false,
    "no-console": true,
    "object-literal-sort-keys": false,
    "max-line-length": [
      true
    ]
  },
  "jsRules": {
    "max-line-length": [
      true
    ]
  }
}
```

#### Actual behavior
```ts
const obj = {};
obj.___id__ = "foobar";  //  three '_' 囧rz
```

#### Expected behavior
```ts
const obj = {};
obj.__id__ = "foobar";
```

**Comment by ajafff** [comment]
There seems to be a bug in typescript's AST. I opened https://github.com/Microsoft/TypeScript/issues/16788 to track this.

**Comment by ajafff** [comment]
Even though that bug is fixed with typescript@2.5.0, I added a fix for older version of typescript: #3184
